strawberryfields/gbs/data.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+GBS Datasets
+============
+
+**Module name:** :mod:`strawberryfields.gbs.data`
+
+.. currentmodule:: strawberryfields.gbs.data
+
+This module provides access to pre-calculated datasets of simulated GBS samples generated from a
+range of input graphs. Each dataset corresponds to a set of samples generated from a fixed graph
+with corresponding values for:
+
+- ``n_mean``: mean number of photons in the GBS device
+- ``n_max``: maximum number of photons allowed in any sample
+-  ``threshold``: flag to indicate whether samples are generated with threshold detection or
+   with photon number resolving detectors.
+- ``n_samples``: total number of samples in dataset
+- ``modes``: number of modes in GBS device or, equivalently, number of nodes in graph
+
+Datasets are available as classes. The following graphs and datasets are provided:
+
+- **Planted graph**: A 30-node graph with a dense 10-node subgraph planted inside
+  :cite:`arrazola2018using`.
+
+  .. image:: ../../_static/planted.png
+        :align: center
+        :width: 300px
+
+  Provided by the :class:`Planted` Dataset with attributes ``n_mean = 6``, ``n_max = 14``,
+  ``threshold = True``, ``n_samples = 50000``, ``modes = 30``.
+
+Loading data
+^^^^^^^^^^^^
+
+We use the :class:`Planted` class as an example to show how to interact with the datasets. Datasets
+can be loaded by running:
+
+>>> data = Planted()
+
+Accessing samples from the dataset is then simple:
+
+>>> i = 3
+>>> sample_i = data[i]
+>>> samples = data[:10]
+
+Datasets also contain metadata relevant to the GBS setup:
+
+>>> data.n_mean
+6
+
+>>> data.n_samples
+50000
+
+Counting photons or clicks in each sample is available using the :meth:`Dataset.counts` method:
+
+>>> data.counts()

```

Comment:
I don't understand this, what is the output of `counts`? Is it the aggregate of counts over all samples in the dataset? For instance, what does `11` refer to in the output of the example usage?